File: csL2/Repo/IRepo.cs
using lab2_mpp.domain;

namespace lab2_mpp.repo;

public interface IRepo<ID, E> where E: Entity<ID>
{
    void Add(E e);
    void Update(E e);
    void Remove(ID id);
    E GetById(ID id);
    IEnumerable<E> GetAll();
    
}

File: javaL2/src/main/java/ro/mpp2024/repo/IRepo.java
package ro.mpp2024.repo;
import ro.mpp2024.domain.Entity;
import ro.mpp2024.repo.exceptions.EntityRepoException;

public interface IRepo<ID, E extends Entity<ID>> {
    E add (E e) throws EntityRepoException;
    E update (E e) throws EntityRepoException;
    E remove (E e) throws EntityRepoException;
    E getById (ID id) throws EntityRepoException;
    Iterable<E> getAll () throws EntityRepoException;
}
